js/layout-javascript/news-feed-code.js

```diff
@@ -2337,99 +2380,126 @@ DynamicList.prototype.sendComment = function(id, value) {
+          var foundUser = _.find(_this.allUsers, function(user) {
+            return user.id === filteredUser.id;
+          });
 
-    if (filteredUsers && filteredUsers.length) {
-      filteredUsers.forEach(function(filteredUser) {
-        var foundUser = _.find(_this.allUsers, function(user) {
-          return user.id === filteredUser.id;
+          if (foundUser) {
+            usersMentioned.push(foundUser);
+          }
         });
+      }
+    }
 
-        if (foundUser) {
-          usersMentioned.push(foundUser);
-        }
+    comment.mentions = [];
+    if (usersMentioned && usersMentioned.length) {
+      usersMentioned.forEach(function(user) {
+        comment.mentions.push(user.id);
       });
     }
-  }
-
-  comment.mentions = [];
-  if (usersMentioned && usersMentioned.length) {
-    usersMentioned.forEach(function(user) {
-      comment.mentions.push(user.id);
-    });
-  }
 
-  comment.text = value;
-  comment.timestamp = timestamp;
+    comment.text = value;
+    comment.timestamp = timestamp;
 
-  return _this.getCommentIdentifier(record)
-    .then(function (identifier) {
-      return Fliplet.Profile.Content({ dataSourceId: _this.data.commentsDataSourceId })
-        .then(function(instance) {
-          return instance.create(identifier, {
-            settings: comment
+    return _this.getCommentIdentifier(record)
+      .then(function (identifier) {
+        return Fliplet.Profile.Content({ dataSourceId: _this.data.commentsDataSourceId })
+          .then(function(instance) {
+            return instance.create(identifier, {
+              settings: comment
+            });
           });
-        });
-    })
-    .then(function(comment) {
-      record.comments.push(comment);
-      _this.replaceComment(guid, comment, 'final');
-    })
-    .catch(function onQueryError(error) {
-      // Reverses count if error occurs
-      console.error(error);
+      })
+      .then(function(comment) {
+        options = {
+          instance: _this,
+          config: _this.data,
+          id: _this.data.id,
+          uuid: _this.data.uuid,
+          container: _this.$container,
+          record: record,
+          commentEntry: comment,
+          commentGuid: guid
+        };
+        return Fliplet.Hooks.run('flListDataAfterNewComment', options)
+          .then(function () {
+            comment = options.commentEntry || comment;
+            record.comments.push(comment);
+            _this.replaceComment(guid, comment, 'final');
+            options.commentContainer = _this.$container.find('.fl-individual-comment[data-id="' + comment.id + '"]');
+            Fliplet.Hooks.run('flListDataAfterNewCommentShown', options);
+          });
+      });
+  }).catch(function (error) {
+    // Reverses count if error occurs
+    console.error(error);
 
-      if (_.get(record, 'commentCount')) {
-        record.commentCount--;
-      }
+    if (_.get(record, 'commentCount')) {
+      record.commentCount--;
```

Comment:
why?

Reply:
The comment count was optimistically incremented. The error caught reverses that.

Note that I'm only unindenting the code here due to changing the position of the `.catch()` clause.